erpnext/manufacturing/doctype/work_order/work_order.py

```diff
@@ -213,6 +213,20 @@ def update_work_order_qty(self):
 
 			self.db_set(fieldname, qty)
 
+			#for multiple work orders against same sales order item
```

Comment:
Make it a function and put it inside sales_order.py, this way you will be able to call the function very easily to correct  the produced_qty in sales order (just like the use case of below patch)

Reply:
sales_order.py or sales_order_item.py? As a single work order is for producing one single item, so produced_qty will be updated in sales_order_item.

Reply:
> sales_order.py or sales_order_item.py? As a single work order is for producing one single item, so produced_qty will be updated in sales_order_item.

Keep it in sales_order.py, we don't write any logic in child table's file.